extensions/jdbc/jdbc-db2/deployment/src/main/java/io/quarkus/jdbc/db2/deployment/JDBCDB2Processor.java

```diff
@@ -55,12 +65,23 @@ void configureAgroalConnection(BuildProducer<AdditionalBeanBuildItem> additional
     }
 
     @BuildStep
-    void registerDriverForReflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+    void registerForReflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         //Not strictly necessary when using Agroal, as it also registers
         //any JDBC driver being configured explicitly through its configuration.
         //We register it for the sake of people not using Agroal,
         //for example when the driver is used with OpenTelemetry JDBC instrumentation.
         reflectiveClass.produce(ReflectiveClassBuildItem.builder(DB2_DRIVER_CLASS).build());
+
+        // register resource bundles for reflection (they are apparently classes...)
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(Resources.class).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(ResourceKeys.class).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(SqljResources.class).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(T2uResourceKeys.class).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(T2uResources.class).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(T2zResourceKeys.class).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(T2zResources.class).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(T4ResourceKeys.class).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(T4Resources.class).build());
```

Comment:
I played around with this, and it seems we also need `oracle.net.mesg.Message`. @gsmet are you ok with me opening a PR and adding that class?